tests/testthat/test-tibble.R

```diff
@@ -497,15 +497,15 @@ test_that("as_tibble() can convert row names for zero-row tibbles", {
   expect_identical(unclass(tbl_df), unclass(df))
 })
 
-test_that("as_tibble() throws an error when user turns missing row names into column", {
+test_that("as_tibble() converts implicit row names when `rownames =` is passed", {
   df <- data.frame(a = 1:3, b = 2:4)
-  expect_tibble_error(
+  expect_equal(
     as_tibble(df, rownames = "id"),
-    error_as_tibble_needs_rownames()
+    tibble(id = as.character(1:3), a = 1:3, b = 2:4)
   )
-  expect_tibble_error(
+  expect_equal(
     as_tibble(df[0, ], rownames = "id"),
-    error_as_tibble_needs_rownames()
+    tibble(id = integer(0), a = integer(0), b = integer(0))
```

Comment:
Does this need to be `character` here?

```suggestion
    tibble(id = character(0), a = integer(0), b = integer(0))
```

Reply:
Yes, you are right. I noticed that. Seems the as.character() is not triggered in this edge case, so the unit test does pass as integer(0), but I don't like that inconsistency either. Let me think about it and re-submit something more elegant/robust.